src/fluree/db/peer/http_api.clj

```diff
@@ -206,19 +206,31 @@
   [_ system param auth-map ledger {:keys [timeout] :as opts}]
   (go-try
     (require-authentication system auth-map)
-    (let [conn        (:conn system)
-          private-key (when (= :jwt (:type auth-map))
-                        (<? (pw-auth/fluree-decode-jwt conn (:jwt auth-map))))
-
-          _           (when-not (sequential? param)
-                        (throw (ex-info (str "A transaction submitted to the 'transact' endpoint must be a list/vector/array.")
-                                        {:status 400 :error :db/invalid-transaction})))
-          txid-only?  (some-> (get opts "txid-only") str/lower-case (= "true"))
-          auth-id     (:auth auth-map)
-          result      (<? (fdb/transact-async conn ledger param {:auth        auth-id
-                                                                 :private-key private-key
-                                                                 :txid-only   txid-only?
-                                                                 :timeout     timeout}))]
+    (let [conn          (:conn system)
+          private-key   (when (= :jwt (:type auth-map))
+                          (<? (pw-auth/fluree-decode-jwt conn (:jwt auth-map))))
+          verified-auth (when (= :http-signature (:type auth-map))
+                          (-> auth-map
+                              (select-keys [:signed :signature])
+                              ;; :authority is guaranteed to be the verified auth-id
+                              ;; derived from the private key that signed the
+                              ;; original request
+                              (assoc :auth (:authority auth-map))))
+          _             (when-not (sequential? param)
+                          (throw (ex-info (str "A transaction submitted to the "
+                                               "'transact' endpoint must be a "
+                                               "list/vector/array.")
+                                          {:status 400
+                                           :error  :db/invalid-transaction})))
+          txid-only?    (some-> opts (get "txid-only") str/lower-case (= "true"))
+          auth-id       (:auth auth-map)
```

Comment:
this is probably outside the scope of this pr, but all the `auth` and `authority` names flying around makes this a bit hard to follow.

Reply:
I agree. Not sure what better names are, and I have `s/db/ledger/g` higher on my priority list before even digging into this one (someday).